Add plugin: Autoescaper

BlueFootedBird opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

Link to my plugin:
https://github.com/bluefootedbird/autoescaper

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [ ]  macOS
  - [x]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [ ] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
Remove `node_modules` from the repo, it should never be in a git repo.

[editorCallback: (editor: Editor) => {](https://github.com/BlueFootedBird/Autoescaper/blob/66ec520191ec64189c017b6ab2f4760386570045/main.ts#L11)
This should be a `editorCheckCallback` so that the command is only available when there is something selected

[name: 'Escape HTML Characters',](https://github.com/BlueFootedBird/Autoescaper/blob/66ec520191ec64189c017b6ab2f4760386570045/main.ts#L10)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)


[console.log(`Escaped highlighted text: ${escapedText}`);](https://github.com/BlueFootedBird/Autoescaper/blob/66ec520191ec64189c017b6ab2f4760386570045/main.ts#L25), [console.log("No text selected, nothing replaced.");](https://github.com/BlueFootedBird/Autoescaper/blob/66ec520191ec64189c017b6ab2f4760386570045/main.ts#L27)

BlueFootedBird commented:
I've addressed the following concerns:

- [x] Removed node_modules
- [x] editorCallback --> editorCheckCallback 
- [x] Used sentence case in the UI
- [x] Refactored console.log

joethei commented:
What is the code you tried with `editorCheckCallback` ?
The function signature is slightly different from the regular `editorCallback`, see: <https://docs.obsidian.md/Reference/TypeScript+API/Command/editorCheckCallback>
